capsules/src/virtual_spi.rs

```diff
@@ -106,6 +106,22 @@ impl<'a, Spi: hil::spi::SpiMaster> MuxSpiMaster<'a, Spi> {
                     Op::Idle => {} // Can't get here...
                 }
             });
+        } else {
+            self.inflight.map(|node| {
+                let op = node.operation.get();
+                // Need to set idle here in case callback changes state
+                node.operation.set(Op::Idle);
```

Comment:
How can this unconditionally set the inflight operation to idle? What if two SPI clients call `read_write_bytes()` back-to-back, and `do_next_op()` is called for the second time?

Reply:
If I'm not mistaking, operation cat be only two values at this point:
- `ReadWriteDone` if this was called from `command_complete`
- `ReadWriteDone` if this is called due to a an error before an action can be started
- `Idle` if this is called when another client tries to schedule an action

But I might be missing something.

Reply:
```suggestion
```